src/uct/tcp/tcp_ep.c

```diff
@@ -615,19 +638,22 @@ static inline unsigned uct_tcp_ep_recv(uct_tcp_ep_t *ep, size_t recv_length)
 static unsigned uct_tcp_ep_progress_data_tx(uct_tcp_ep_t *ep)
 {
     unsigned count = 0;
-    size_t sent_length;
+    ssize_t ret;
 
     ucs_trace_func("ep=%p", ep);
 
     if (uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
-        if (!(ep->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_ZCOPY_TX))) {
-            count += uct_tcp_ep_send(ep, &sent_length);
-        } else {
-            count += uct_tcp_ep_sendv(ep, &sent_length);
+        ret = (!(ep->ctx_caps & UCS_BIT(UCT_TCP_EP_CTX_TYPE_ZCOPY_TX)) ?
+               uct_tcp_ep_send(ep) : uct_tcp_ep_sendv(ep));
+        if (ucs_unlikely(ret < 0)) {
+            uct_tcp_ep_handle_disconnected(ep, &ep->tx);
+            return 1;
         }
 
-        ucs_trace_data("ep %p fd %d sent %zu/%zu bytes, moved to offest %zu",
-                       ep, ep->fd, ep->tx.offset, ep->tx.length, sent_length);
+        count += (ret > 0);
```

Comment:
minor: seems like `count`, actually, is not a counter and this var can be just initialized here. maybe rename `ret` ->  `offset` like it's mentioned in trace below and `count` -> `ret` ?

Reply:
good point - done